app/frontend/components/pages/receiveAda/addressItem.js

```diff
@@ -1,68 +1,109 @@
-const {h} = require('preact')
+const {h, Component} = require('preact')
 const connect = require('unistore/preact').connect
 const actions = require('../../../actions')
 
 const CopyOnClick = require('../../common/copyOnClick')
 
-const AddressItem = connect(
-  {},
-  actions
-)(({address, bip32path, openAddressDetail}) =>
-  h(
-    'div',
-    {class: 'address'},
-    h(
-      'div',
-      {class: 'address-value'},
-      h('p', {class: 'address-number no-select'}, `/${bip32path.split('/').pop()}`),
-      h('p', {class: 'one-click-select'}, address)
-    ),
-    h(
+class AddressItem extends Component {
+  constructor(props) {
+    super(props)
+    this.updateDimensions = this.updateDimensions.bind(this)
+    this.openAddressDetail = this.openAddressDetail.bind(this)
+  }
+
+  openAddressDetail({address, bip32path}, copyOnClick) {
+    this.props.openAddressDetail({address, bip32path}, copyOnClick)
+  }
+
+  updateDimensions() {
+    if (window.innerWidth < 768) {
+      this.setState({onSmallDevice: true})
+    } else if (window.innerWidth > 767) {
```

Comment:
this is a little bit strange, why not check for <= 767 in the first 'if' and have just 'else' instead of 'else if'?

Reply:
You are right, dont know how that happened, :smile:  it is strange, fixed.